packages/web-frontend/src/containers/MeasureLegend/MultiLegend.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+import { MeasureOptionsGroupPropType } from '../../components/Marker/propTypes';
+import { BOX_SHADOW, OFF_WHITE, TRANS_BLACK_LESS } from '../../styles';
+import {
+  MEASURE_TYPE_COLOR,
+  MEASURE_TYPE_ICON,
+  MEASURE_TYPE_RADIUS,
+  MEASURE_TYPE_SHADED_SPECTRUM,
+  MEASURE_TYPE_SPECTRUM,
+} from '../../utils/measures';
+import { MarkerLegend } from './MarkerLegend';
+import { SpectrumLegend } from './SpectrumLegend';
+
+const LegendOuterFrame = styled.div`
+  width: 100%;
+  display: flex;
+`;
+
+const LegendFrame = styled.div`
+  padding: 5px;
+  margin: 10px auto;
+  border-radius: 3px;
+  cursor: auto;
+  color: ${OFF_WHITE};
+  background-color: ${TRANS_BLACK_LESS};
+  box-shadow: ${BOX_SHADOW};
+`;
+
+const coloredMeasureTypes = [
+  MEASURE_TYPE_COLOR,
+  MEASURE_TYPE_SPECTRUM,
+  MEASURE_TYPE_SHADED_SPECTRUM,
+];
+
+const NullLegend = () => null;
+
+const getLegend = measureType => {
+  switch (measureType) {
+    case MEASURE_TYPE_SHADED_SPECTRUM:
+    case MEASURE_TYPE_SPECTRUM:
+      return SpectrumLegend;
+    case MEASURE_TYPE_RADIUS:
+      return NullLegend;
+    default:
+      return MarkerLegend;
+  }
+};
+
+// returning <LegendOuterFrame /> keeps the map control on the right hand side
+const renderEmptyMultiLegend = () => <LegendOuterFrame />;
+
+const MultiLegend = ({ measureOptions, isMeasureLoading }) => {
+  if (isMeasureLoading) {
```

Comment:
```suggestion
const displayedLegends = measureOptions.filter(({ type }) => type !== MEASURE_TYPE_RADIUS);

  if (isMeasureLoading || !displayedLegends) {
    return renderEmptyMultiLegend();
  }
```

Reply:
It would be great if possible to handle both empty legend cases at once at the top. measureOptions.filter should always be safe to call because measureOptions is a required field in propTypes. If it causes errors, then you could update measureOptions to have a defaultPropType = [] or something;

Reply:
Always love a diff with more minus than plus! And measureOptions is defaulted to [] in mapStateToProps, so that should be no issue :).